### PR TITLE
docs: fix typos

### PR DESCRIPTION
**Pull request by ucpr**
I found some typos in the docs and fixed them.

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ Gatling Runner Pod runs multiple containers. The diagram below shows how those c
 - gatling-runner
   - Executes Gatling load testing scenario
   - Stores its generated report file (`simulation.log`) to an emptyDir volume
-- gatling-result-transferer
+- gatling-result-transfer
```
Comment (yokawasa) [inline docs/architecture.md:43]: The original one (`transferer`) should be ok.  We named it`transferer` to mean the one who transfer something 

```suggestion
- gatling-result-transferer
```

**Review by yokawasa — COMMENTED** [review]
@ucpr thanks for the contribution! ❤️ 
it generally looks great to me except one thing (please check the suggested comment)

File: docs/architecture.md (hunk 1 above)
Comment (ucpr) [reply docs/architecture.md]: Thank you. Fixed!
I found `gatling-result-transfer` elsewhere and fixed it.

https://github.com/st-tech/gatling-operator/pull/56/commits/6b54aee130f273f49d4f0c5645ec508c1d1adc33

**Comment by yokawasa** [comment]
@ucpr LGTM, thanks for fixing!!

**Review by yokawasa — APPROVED** [review]

**Pull request merged**